### DISCUSSION TITLE
Abort Button in Laser GRBL doesnt clear buffer

### DISCUSSION THREAD

**Discussion by jomo487** [arkypita/LaserGRBL#1731] [Q&A] [answered]
When the abort button in Lasergrbl is pressed the  laser continues to move and burn until the buffer is emptied. It is difficult to reset and restart a project. Usually disconnecting and closing Lasergrbl, unplugging usb and restarting Lasergrbl.  The Halt button does stop it right away, but is also difficult to start over again.  
Win 7 , lasergrbl 4.7.1, firmware from lasergrbl TOOLS button
Any Thoughts would be welcome.

**Comment by arkypita** [comment, marked as answer]
Yes, and this is by design.

Once lasergrbl has sent the gcode to grbl firmware, the queued lines are inside the controller, and outside of lasergrbl control.
The abort button (i think you are talking about the squared red one) simply stop sending new gcode lines, but nothing can do to "remove" the already sent lines, so the engraver stop when the controller finish to process all the pending ones.

The only "immediate" abort is the reset option (thunderbolt button) that reset the controller.

Grbl firmare does not provide any better option (for what I know).

**Reply by jomo487** [reply]
Thank you
Yes the Reset button is a much cleaner option for terminating a bad run. 
Are there any reasons the halt button is preferable? I would think shutting the laser off immediately is more important when there is a problem.

By the way, Great Program! thanks for your and your contributors efforts and support.
